.github/workflows/pull-request-checks.yaml

```diff
@@ -4,6 +4,42 @@ on:
     branches: [ develop ]
 
 jobs:
+  check-ubuntu-20_04-make-gcc:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Fetch dependencies
+        env:
+          # This is needed in addition to -yq to prevent apt-get from asking for
+          # user input
+          DEBIAN_FRONTEND: noninteractive
+          TESTPL_JOBS: 4
+        run: |
+          sudo apt-get install -yq gcc gdb g++ maven jq flex bison libxml2-utils cpanminus
+          make -C src minisat2-download
```

Comment:
I'd be more inclined to move this `make` invocation into the build section - it's part of the system-under-tests (i.e. the Makefiles) rather than general environment setup (the apt-get)

Reply:
Yes, but afaik this doesn't really build minisat, does it? It only downloads the sources and applies a patch as far as I can recall?

Reply:
All of which is defined in the Makefile, which is what is being tested here.

Reply:
Yes, I recall correctly (https://github.com/diffblue/cbmc/blob/develop/src/Makefile#L120).

I would argue this is part of the configuration, as without minisat being downloaded, the build fails - but it's really because the sources are missing, not because of a compilation issue or anything else. And seeing as minisat is one of the SAT solvers we support, I would argue even more this is part of the configuration, even if it's just the default right now.

Reply:
I don't have a massive issue with this - so I'm happy to approve if everyone else is happy. My logic was:

1) Cmake does the minisat download as part of its configure/build steps, so my suggestion would be more consistent with that
2) If the apt-get fails, or the cpanm fails, its mostly just a case of saying "shrug, infrastructure failed - hit retry", but if the minisat-download fails you _do_ have to double check whether our own stuff needs to be modified. If you have the "setup our CI host" as a separate section to "run our own stuff" then it makes a clear distinction between stuff we may need to manage/modify and stuff that is just pure environment setup.